src/main/java/org/cactoos/time/DateOf.java

```diff
@@ -0,0 +1,197 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.time;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import org.cactoos.Func;
+import org.cactoos.Scalar;
+import org.cactoos.scalar.UncheckedScalar;
+
+/**
+ * Parser for date instances.
+ * @author Sven Diedrichsen (sven.diedrichsen@gmail.com)
+ * @version $Id$
+ * @param <T> Type of the date to parse to.
+ * @since 1.0
+ */
+public class DateOf<T extends TemporalAccessor> implements Scalar<T> {
+
+    /**
+     * The parsed date.
+     */
+    private UncheckedScalar<T> parsed;
+
+    /**
+     * Parses the date using the provided format.
+     * @param date The date to parse.
+     * @param format The format to use.
+     * @param converter The converter to convert
+     *  {@link TemporalAccessor} into type T
+     */
+    public DateOf(final String date, final String format,
+        final Func<TemporalAccessor, T> converter) {
+        this(date, DateTimeFormatter.ofPattern(format), converter);
+    }
+
+    /**
+     * Parses the provided date as ISO formatted.
+     * @param date The date to parse.
+     * @param converter The converter to convert
+     *  {@link TemporalAccessor} into type T
+     */
+    public DateOf(final String date,
+        final Func<TemporalAccessor, T> converter) {
+        this(date, DateTimeFormatter.ISO_DATE_TIME, converter);
+    }
+
+    /**
+     * Parsing the date using the provided formatter.
+     * @param date The date to parse.
+     * @param formatter The formatter to use.
+     * @param converter The converter to convert
+     *  {@link TemporalAccessor} into type T
+     */
+    public DateOf(final String date, final DateTimeFormatter formatter,
+        final Func<TemporalAccessor, T> converter) {
+        this.parsed = new UncheckedScalar<>(
+            () -> converter.apply(formatter.parse(date))
+        );
+    }
+
+    @Override
+    public final T value() throws Exception {
+        return this.parsed.value();
+    }
+
+    /**
+     * Convenience class for parsing to {@link LocalDateTime} instances.
+     */
+    public static class Local extends DateOf<LocalDateTime> {
```

Comment:
@svendiedrichsen where is `DateOf.Date`? I need instances of `Date`. I believe, not just me.

Reply:
I have added the DateOf.Date class. It can't derive from DateOf 'cause it doesn't extends TemporalAccessor class.